content/docs/connectors.md

```diff
@@ -0,0 +1,634 @@
+                    "type": "string"
+                  },
+                  "collection": {
+                    "deprecated": false,
+    ...
+                    "type": "string"
+                  },
+                  "operation": {
+                    "deprecated": false,
+    ...
+                    "type": "string"
+                  }              
+                }
+            }    
+            ]
+        }    
+      ],
+
+The second part of the descriptor is where we introduce to provide the expected behavior of our component. We can define “actions” that are going to be performed over the connection according to the pattern specified (mainly `From` for a consumer and `To` for a producer type - also `Pipe` exists and can be configured).
+
+Here we just want to use the connector as a producer of data in order to perform certain operations as final result of a route. As an example, we want to get data from a certain source (a sql db, your twitter mentions, …) and write to a mongoDB database and collection.
+
+In order to perform such operation I need to know the following 3 parameters: `database`, `collection` and `operation`. Therefore in the `propertiesDefinitionStep` we include them. The runtime result will be that we’re going to be prompted from the GUI to provide those configuration every time that we’ll create a new integration on Syndesis.
+
+      "dependencies": [
+        {
+          "id": "@project.groupId@:@project.artifactId@:@project.version@",
+          "type": "MAVEN"
+        }
+      ]
+    }
+
+The last part of the descriptor is used to declare maven dependencies. The above one should work from most of the case if the development conventions are respected.
+### Unit test preparation
+At this stage we have our connector, so, we can even try to include in Syndesis bundle, deploy and make a real integration. However, a better strategy is to make some unit test in order to verify that our integration is not going to fail and to serve as regression for the future.
+
+There is a class you have to extend in order to benefit all the environment provided by Syndesis and Camel and just worry about making your testsuite: [io.syndesis.connector.support.test.ConnectorTestSupport](https://github.com/syndesisio/syndesis/blob/master/app/connector/support/test/src/main/java/io/syndesis/connector/support/test/ConnectorTestSupport.java) that is extending the upstream Camel supporting test class [org.apache.camel.test.junit4.CamelTestSupport](https://github.com/apache/camel/blob/master/components/camel-test/src/main/java/org/apache/camel/test/junit4/CamelTestSupport.java).
+
+As far as you progress with the development of your test suit you will be able to discover all the features offered by these 2 classes. For the first iteration you will have to just provide the implementation of an abstract method that would take care of building the final route (involving your connector too!). Here is a simple test for the MongoDB:
```

Comment:
"The more you progress"... better? "As far as" sounds to me like a false friend.

Reply:
Probably "as long as" would be better.